src/renderer/store/transcriptionWords/reducer.ts

```diff
@@ -132,6 +136,25 @@ const transcriptionWordsReducer: Reducer<Word[], Action<any>> = (
     );
   }
 
+  if (action.type === RESTORE_SECTION) {
+    const { ranges } = action.payload as RestoreSectionPayload;
+
+    const markUndeleted = (word: Word) => ({
+      ...word,
+      deleted: false,
+    });
+
+    return mapInRanges(words, markUndeleted, ranges);
+  }
+
+  if (action.type === UNDO_RESTORE_SECTION) {
+    const { ranges } = action.payload as UndoRestoreSectionPayload;
+
+    const markDeleted = (word: Word) => ({ ...word, deleted: true });
```

Comment:
I feel like these are used enough they should be extracted as utils

Reply:
Done.